test/e2e/set_delete_test.go

```diff
@@ -0,0 +1,235 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+var podCount int
+
+func testHeaders(t *testing.T, image string, route *routev1.Route, address string, headers []string, expectedResponse string, expectedMatches int) {
+	t.Helper()
+
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("failed to get kube config: %v", err)
+	}
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("failed to create kube client: %v", err)
+	}
+
+	var extraCurlArgs []string
+	for _, header := range headers {
+		extraCurlArgs = append(extraCurlArgs, "-H", header)
+	}
+	extraCurlArgs = append(extraCurlArgs, "-v", "--resolve", route.Spec.Host+":80:"+address)
+	podCount++
+	name := fmt.Sprintf("%s%d", route.Name, podCount)
+	clientPod := buildCurlPod(name, route.Namespace, image, route.Spec.Host, address, extraCurlArgs...)
+	if err := kclient.Create(context.TODO(), clientPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), clientPod); err != nil {
+			if !errors.IsNotFound(err) {
+				t.Fatalf("failed to delete pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+			}
+		}
+	}()
+	expectedResponse = strings.ToLower(expectedResponse)
+	err = wait.PollImmediate(1*time.Second, 4*time.Minute, func() (bool, error) {
```

Comment:
This is failing because the pod, service, route are defer-deleted before you get a chance to access the logs.  See https://github.com/openshift/cluster-ingress-operator/pull/872/files#r1289168086 and https://github.com/openshift/cluster-ingress-operator/pull/872/files#r1289168463.

Reply:
This is incorrect, please disregard.